evap/results/views.py

```diff
@@ -340,3 +331,65 @@ def add_warnings(evaluation, evaluation_result):
 
         for rating_result in rating_results:
             rating_result.warning = questionnaire_result.warning or rating_result.has_answers and rating_result.count_sum < questionnaire_warning_thresholds[questionnaire_result.questionnaire]
+
+
+def evaluation_detail_parse_get_parameters(request, evaluation):
+    if not evaluation.can_results_page_be_seen_by(request.user):
+        raise PermissionDenied
+
+    if request.user.is_reviewer:
+        view = request.GET.get('view', 'public')  # if parameter is not given, show public view.
+    else:
+        view = request.GET.get('view', 'full')  # if parameter is not given, show own view.
+    if view not in ['public', 'full', 'export']:
+        view = 'public'
+
+    view_as_user = request.user
+    contributor_id = int(request.GET.get('contributor_id', request.user.id))
+    if view == 'export' and request.user.is_staff:
+        view_as_user = UserProfile.objects.get(id=contributor_id)
+    contributor_id = contributor_id if contributor_id != request.user.id else None
+
+    represented_users = [view_as_user]
+    if view != 'export':
+        represented_users += list(view_as_user.represented_users.all())
+    # redirect to non-public view if there is none because the results have not been published
+    if not evaluation.can_publish_rating_results and view == 'public':
+        view = 'full'
+
+    return view, view_as_user, represented_users, contributor_id
+
+
+def extract_evaluation_answer_data(request, evaluation):
+    # TextAnswerExcelExporter wants a dict from Question to tuple of contributor_name and string list (of the answers)
+
+    view, view_as_user, represented_users, contributor_id = evaluation_detail_parse_get_parameters(request, evaluation)
+
+    evaluation_result = collect_results(evaluation)
+    filter_text_answers(evaluation_result)
+    remove_textanswers_that_the_user_must_not_see(evaluation_result, view_as_user, represented_users, view)
+
+    results = TextAnswerExcelExporter.InputData(evaluation_result.contribution_results)
+
+    return results, contributor_id
+
+
+@reviewer_required
```

Comment:
please remove this, we decided the correct test (`reviewer_or_responsible_or_contributor_or_delegate_required`) which does not exist yet might be too much.